src/abstractdatatable/join.jl

```diff
@@ -3,9 +3,15 @@
 ##
 
 # Like similar, but returns a nullable array
+similar_nullable{T}(dv::AbstractArray{T}, dims::@compat(Union{Int, Tuple{Vararg{Int}}})) =
```

Comment:
This commit just reverts changes made in the previous commits (from the other PR). Nothing changes overall. @cjprybol I think it would be better to focus on removing `readtable` first, since that's simple, and then you can rebase this PR so that only commits related to it are shown. Else it's really messy.